File: 13-RomanToInteger/13-RomanToInteger.java
// Last updated: 9/8/2025, 4:38:41 PM
class Solution {
    public int romanToInt(String s) {
        int total = 0;
        String roman = s+"0";
        int i = 0;
        while (roman.charAt(i) != '0'){
            switch(roman.charAt(i)){
                case 'I':
                    if (roman.charAt(i+1) == 'V'){
                        total += 4;
                        i++;
                    } 
                    else if (roman.charAt(i+1) == 'X'){
                        total += 9;
                        i++;
                    }
                    else{
                        total+=1;
                    }
                break;
                case 'V':
                    total+=5;
                break;
                case 'X':
                    if (roman.charAt(i+1) == 'L'){
                        total += 40;
                        i++;
                    } 
                    else if (roman.charAt(i+1) == 'C'){
                        total += 90;
                        i++;
                    } 
                    else{

                        total+=10;
                    }
                break;
                case 'L':
                    total += 50;
                break;
                case 'C':
                    if (roman.charAt(i+1) == 'D'){
                        total += 400;
                        i++;
                    } 
                    else if (roman.charAt(i+1) == 'M'){
                        total += 900;
                        i++;
                    } 
                    else{

                        total+=100;
                    }
                break;
                case 'D':
                    total += 500;
                break;
                case 'M':
                    total += 1000;
                break;

                case '0':
                break;
            } 
        i++;
        }
    return total;
    }
}

File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 2/15/2026, 8:54:26 PM
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_dict = {
            'I': 1,
            'V': 5, 
            'X': 10,
            'L': 50, 
            'C': 100, 
            'D': 500, 
            'M': 1000
        }
        total = 0
        prev = ''
        for i in range(0, len(s)):
            total += roman_dict[s[i]]
            if ((s[i] == 'V' or s[i] == 'X') and prev == 'I'):
                total -= 2
            elif ((s[i] == 'L' or s[i] == 'C') and prev == 'X'):
                total -= 20
            elif ((s[i] == 'D' or s[i] == 'M') and prev == 'C'):
                total -= 200
            prev = s[i]
        return total